token/program-2022/src/extension/mod.rs

```diff
@@ -404,20 +404,20 @@ pub trait BaseStateWithExtensions<S: BaseState> {
     ///
     /// Provides the correct answer regardless if the extension is already present
     /// in the TLV data.
-    fn try_get_new_account_len<V: Extension + VariableLenPack>(
+    fn try_get_new_account_len<V: Extension>(
```

Comment:
This is kind of a footgun now, since you can call `try_get_new_account_len` with a fixed-length extension and pass the wrong size.

How about making two versions, one for fixed-length, and one for variable-length, and have them call into a common private function? The variable length one can change to take in the new size if you think that's clearer

Reply:
This is what I had, believe it or not, but I wanted to minimize changes to this code. But I can return to that!